sources/Renderer/ModuleInterface.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * GLModuleInterface.h
+ * 
+ * This file is part of the "LLGL" project (Copyright (c) 2015-2018 by Lukas Hermanns)
+ * See "LICENSE.txt" for license information.
+ */
+
+#include "ModuleInterface.h"
+#include <LLGL/RenderSystemFlags.h>
+
+#ifdef LLGL_BUILD_STATIC_LIB
+
+namespace LLGL
+{
+    
+#ifdef LLGL_HAS_MODULE_DX11
```

Comment:
Where are these macros defined? I can't find them in the `CMakeLists.txt` file.

Reply:
I use a custm build system. I'm not a familiar with cmake so i prefer to leave to someone else to do it instead of me breaking something there.

Reply:
Alright, I can take care of that.
Just out of curiosity: What kind of custom build system do you have?

Reply:
I use python to collect resources and i finally generate a premake5 "solution". Out of that i run premake5 to generate solution for xcode/vs/linux. It stuck with me because describing what and how to build tends to be a lot shorter and more flexible than cmake. For example llgl project only takes 130 lines of python code. Also there are some limitations with cmake that i previously ran into (compilation units, precompiled headers, magic errors, versions mismatch, etc). The code is not available publicly but you can see an example project here: https://github.com/raxvan/cllio/blob/master/prj.cllio.py

Reply:
Would be interesting to see alternative build systems for LLGL in the future, but it's probably more work to maintain multiple build systems at the same time. I once saw a [talk from Unity](https://www.youtube.com/watch?v=jmadc8xI_6I) about the build system Gradle. I never came to transition to this system though.